zboxcore/sdk/sdk.go

```diff
@@ -886,6 +892,43 @@ func CreateAllocationForOwner(owner, ownerpublickey string,
 	return
 }
 
+func getAllocationBlobbers(owner, ownerpublickey string,
+	datashards, parityshards int, size, expiry int64,
+	readPrice, writePrice PriceRange, mcct time.Duration) ([]string, error) {
+
+	var allocationRequest = map[string]interface{}{
+		"data_shards":                   datashards,
+		"parity_shards":                 parityshards,
+		"size":                          size,
+		"owner_id":                      owner,
+		"owner_public_key":              ownerpublickey,
+		"expiration_date":               expiry,
+		"read_price_range":              readPrice,
+		"write_price_range":             writePrice,
+		"max_challenge_completion_time": mcct,
+		"diversify_blobbers":            false,
+	}
+
+	allocationData, _ := json.Marshal(allocationRequest)
+
+	params := make(map[string]string)
+	params["allocation_data"] = string(allocationData)
+	params["max_challenge_time"] = mcct.String()
+
+	allocBlobber, err := zboxutil.MakeSCRestAPICall(STORAGE_SCADDRESS, "/get_alloc_blobbers", params, nil)
```

Comment:
The call to `get_alloc_blobbers` should pass exactly the parameters that are required.

-  number of blobbers required
- size of allocation
- expiration
- min read price
- max read price
- min write price
- max write price
- max challenge completion time

The number of blobbers required is to be calculated as
```
number of blobbers = data_shards + parity shards - number of preferred blobbers
```
If the number of preferred blobbers is >= data shards + parity shards, then we can dispense with the get_alloc_blobbers" call.

Reply:
@Sriep Yes, the number of parameters were same as what are required for the old allocation request, but the flow was not working, hence right now the parameters are just minimal to actually see if gosdk & zbox are able to get the blobbers & populate them as prefferedBlobbers.